File: contains-duplicate-two/ContainsDuplicateTwo.swift
class ContainsDuplicateTwo {
     func containsNearbyDuplicate(_ nums: [Int], _ k: Int) -> Bool {
        var set: Set<Int> = []
        for i in 0 ..< nums.count {
            if(set.contains(nums[i])){ return true }
            set.insert(nums[i])
            if(set.count > k) {
                set.remove(nums[i - k])
            }
        }
        return false   
    }
}

File: contains-duplicate-two/ContainsDuplicateTwo.cs
using System.Collections.Generic;

public class ContainsDuplicateTwo{
    public bool ContainDuplicateTwo(int[] nums, int k) {
        HashSet<int> set = new HashSet<int>();
        for(int i = 0; i < nums.Length; i++) {
            if(set.Contains(nums[i])) { 
                return true; 
            }
            set.Add(nums[i]);
            if(set.Count > k) {
                set.Remove(nums[i - k]);
            }
        }
        return false;   
    }
}